### PR TITLE
feat: add logs

### PR DESCRIPTION
**Pull request by pichoemr**
#__ 

## Checklist
- [ ] I didn't over-scope my PR
- [ ] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I did not include breaking changes
- [ ] I included unit tests that cover my changes
- [ ] I added/updated the documentation about my changes
- [ ] I made my own code-review before requesting one

## Technical highlight/advice

https://github.com/lenra-io/lenra-common/pull/15

### CODE REVIEW

File: lib/channels/app_socket.ex [hunk 1]
```diff
@@ -51,7 +51,11 @@ defmodule ApplicationRunner.AppSocket do
                create_metadatas(user_id, app_name, context),
              start_time <- Telemetry.start(:app_session, session_metadata),
              {:ok, session_pid} <- Session.start_session(session_metadata, env_metadata) do
-          Logger.info("joined app #{app_name} with params #{inspect(params)}")
+          Logger.notice("joined app #{app_name} with params #{inspect(params)}")
```
Comment (jonas-martinez) [inline lib/channels/app_socket.ex]:
```suggestion
          Logger.notice("Joined app #{app_name} with params #{inspect(params)}")
```

File: lib/channels/route_channel.ex [hunk 2]
```diff
@@ -75,26 +86,32 @@ defmodule ApplicationRunner.RouteChannel do
...
       def handle_info({:send, :error, {:error, err}}, socket) when is_struct(err) do
-        Logger.error("Send error #{inspect(err)}")
+        # Log a debug, normally errors are logged before sending the error to the channel.
+        Logger.debug("Send error #{inspect(err)}")
 
         push(socket, "error", ErrorHelpers.translate_error(err))
         {:noreply, socket}
       end
 
+      # Send an error if the ui are malformated
```
Comment (jonas-martinez) [inline lib/channels/route_channel.ex]:
```suggestion
      # Send an error if the ui is malformed
```

File: lib/environment/dynamic_supervisor.ex [hunk 3]
```diff
@@ -33,11 +38,19 @@ defmodule ApplicationRunner.Environment.DynamicSupervisor do
   @spec start_env(term()) ::
           {:error, {:already_started, pid()}} | {:ok, pid()} | {:error, term()}
   def start_env(env_metadata) do
+    Logger.debug(
+      "#{__MODULE__} Start Environment Supervisor with env_metadta: #{inspect(env_metadata)}"
```
Comment (jonas-martinez) [inline lib/environment/dynamic_supervisor.ex]:
```suggestion
      "#{__MODULE__} Start Environment Supervisor with env_metadata: #{inspect(env_metadata)}"
```

File: lib/environment/dynamic_supervisor.ex [hunk 4]
```diff
@@ -33,11 +38,19 @@ defmodule ApplicationRunner.Environment.DynamicSupervisor do
...
+    )
+
     case DynamicSupervisor.start_child(
            __MODULE__,
            {ApplicationRunner.Environment.Supervisor, env_metadata}
          ) do
       {:error, {:shutdown, {:failed_to_start_child, _module, reason}}} ->
+        Logger.critical(
+          "#{__MODULE__} failed to start Environment Supervisor with env_metadta: #{inspect(env_metadata)} for reason: #{inspect(reason)}"
```
Comment (jonas-martinez) [inline lib/environment/dynamic_supervisor.ex]:
```suggestion
          "#{__MODULE__} failed to start Environment Supervisor with env_metadata: #{inspect(env_metadata)} for reason: #{inspect(reason)}"
```

File: lib/environment/dynamic_supervisor.ex [hunk 5]
```diff
@@ -64,13 +86,16 @@ defmodule ApplicationRunner.Environment.DynamicSupervisor do
   """
   @spec stop_env(number()) :: :ok | {:error, LC.BusinessError.t()}
   def stop_env(env_id) do
+    Logger.debug("Stop Environment for env_id: #{env_id}")
```
Comment (jonas-martinez) [inline lib/environment/dynamic_supervisor.ex]:
```suggestion
    Logger.debug("Stopping environment for env_id: #{env_id}")
```

File: lib/environment/dynamic_supervisor.ex [hunk 6]
```diff
@@ -64,13 +86,16 @@ defmodule ApplicationRunner.Environment.DynamicSupervisor do
...
     name = Environment.Supervisor.get_name(env_id)
 
     case Swarm.whereis_name(name) do
       :undefined ->
+        Logger.error("Failed to found Supervision Tree for name: #{inspect(name)}")
```
Comment (jonas-martinez) [inline lib/environment/dynamic_supervisor.ex]:
```suggestion
        Logger.error("Failed to find supervision tree for name: #{inspect(name)}")
```

File: lib/environment/dynamic_supervisor.ex [hunk 7]
```diff
@@ -64,13 +86,16 @@ defmodule ApplicationRunner.Environment.DynamicSupervisor do
...
         BusinessError.env_not_started_tuple()
 
       pid ->
+        Logger.info("Stop Environment Supervision Tree for name: #{inspect(name)}")
```
Comment (jonas-martinez) [inline lib/environment/dynamic_supervisor.ex]:
```suggestion
        Logger.info("Stopping environment supervision tree for name: #{inspect(name)}")
```

File: lib/environment/manifest_handler.ex [hunk 8]
```diff
@@ -25,6 +31,9 @@ defmodule ApplicationRunner.Environment.ManifestHandler do
     end
   end
 
+  @doc """
+   Return the Manifest for the given env_id
```
Comment (jonas-martinez) [inline lib/environment/manifest_handler.ex]:
```suggestion
   Returns the Manifest for the given env_id
```

File: lib/environment/metadata.ex [hunk 9]
```diff
@@ -1,6 +1,6 @@
 defmodule ApplicationRunner.Environment.Metadata do
   @moduledoc """
-    The Action struct.
+    The Environmnet metadata.
```
Comment (jonas-martinez) [inline lib/environment/metadata.ex]:
```suggestion
    The Environment metadata.
```

File: lib/environment/mongo_instance.ex [hunk 10]
```diff
@@ -1,16 +1,21 @@
...
   @moduledoc """
-    This module provide the config option to start the `Mongo` genserver in the environment supervisor.
+    This module provides some tools to manage `Mongo` genserver.
   """
   require Logger
 
   use SwarmNamed
 
+  @doc """
+    Return the config option to start the `Mongo` genserver in the environment supervisor, for the given env_id.
```
Comment (jonas-martinez) [inline lib/environment/mongo_instance.ex]:
```suggestion
    Returns the config options to start the `Mongo` genserver in the environment supervisor, for the given env_id.
```

File: lib/environment/query_server.ex [hunk 11]
```diff
@@ -87,20 +68,52 @@ defmodule ApplicationRunner.Environment.QueryServer do
...
+    group = group_name(session_id)
+    Swarm.join(group, pid)
+  end
+
+  def get_data(env_id, coll, query_parsed) do
+    GenServer.call(get_full_name({env_id, coll, query_parsed}), :get_data)
+  end
+
+  @doc """
+    Start monotoring the given ViewServer
```
Comment (jonas-martinez) [inline lib/environment/query_server.ex]:
```suggestion
    Start monitoring the given ViewServer
```

File: lib/environment/view_server.ex [hunk 12]
```diff
@@ -64,22 +67,32 @@ defmodule ApplicationRunner.Environment.ViewServer do
     end
   end
 
+  @doc """
+    Receive notifcation from QueryServer when data changed and we need to refresh view.
```
Comment (jonas-martinez) [inline lib/environment/view_server.ex]:
```suggestion
    Receive notification from QueryServer when data changed and we need to refresh the view.
```

File: lib/errors/alert_agent.ex [hunk 13]
```diff
@@ -0,0 +1,32 @@
+defmodule ApplicationRunner.Errors.AlertAgent do
+  @moduledoc """
+    LenraCommon.Erros.AlertAgent manage Alert & Emergency error
```
Comment (jonas-martinez) [inline lib/errors/alert_agent.ex]:
```suggestion
    ApplicationRunner.Errors.AlertAgent manages Alert & Emergency errors
```

File: lib/errors/alert_agent.ex [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    Agent.start_link(fn -> alert_metadata end, name: {:via, :swarm, error})
+  end
+
+  def send_alert(agent, error) do
+    Agent.update(agent, fn alert_metadata ->
+      # If we have already sent less than 5 errors, send a new alert.
+      if alert_metadata.alert_sent <= 5 do
+        Logger.alert(error)
+        Map.update(alert_metadata, :alert_sent, 1, fn value -> value + 1 end)
+        # if we send 5 alerts, send an emergency and reset the counter.
```
Comment (jonas-martinez) [inline lib/errors/alert_agent.ex]:
```suggestion
        # if we sent 5 alerts, send an emergency and reset the counter.
```

File: lib/errors/technical_error.ex [hunk 15]
```diff
@@ -16,6 +16,7 @@ defmodule ApplicationRunner.Errors.TechnicalError do
       {:openfaas_not_reachable, "Openfaas could not be reached."},
       {:timeout, "Timeout"},
       {:listener_not_found, "Listener not found"},
-      {:mongo_error, "Could not access mongo. Please retry later."}
+      {:mongo_error, "Could not access mongo. Please retry later."},
```
Comment (jonas-martinez) [inline lib/errors/technical_error.ex]:
```suggestion
      {:mongo_error, "Could not access mongo. Please try again later."},
```

File: lib/guardian/app_guardian.ex [hunk 16]
```diff
@@ -45,12 +47,27 @@ defmodule ApplicationRunner.Guardian.AppGuardian do
   defp get_app_token(claims) do
     case claims["type"] do
       "session" ->
-        Session.fetch_token(claims["sub"])
+        try do
+          Session.fetch_token(claims["sub"])
+        rescue
+          e ->
+            Logger.error(
+              "#{__MODULE__} faile to fetch session token for claims: #{inspect(claims)} with error: #{inspect(e)}"
```
Comment (jonas-martinez) [inline lib/guardian/app_guardian.ex]:
```suggestion
              "#{__MODULE__} failed to fetch session token for claims: #{inspect(claims)} with error: #{inspect(e)}"
```

File: lib/guardian/app_guardian.ex [hunk 17]
```diff
@@ -45,12 +47,27 @@ defmodule ApplicationRunner.Guardian.AppGuardian do
...
+        end
 
       "env" ->
-        Environment.fetch_token(String.to_integer(claims["sub"]))
+        try do
+          Environment.fetch_token(String.to_integer(claims["sub"]))
+        rescue
+          e ->
+            Logger.error(
+              "#{__MODULE__} faile to fetch environment token for claims: #{inspect(claims)} with error: #{inspect(e)}"
```
Comment (jonas-martinez) [inline lib/guardian/app_guardian.ex]:
```suggestion
              "#{__MODULE__} failed to fetch environment token for claims: #{inspect(claims)} with error: #{inspect(e)}"
```

File: lib/guardian/app_guardian.ex [hunk 18]
```diff
@@ -45,12 +47,27 @@ defmodule ApplicationRunner.Guardian.AppGuardian do
...
+            )
+        end
 
       _err ->
+        Logger.error("#{__MODULE__} Unknow token type for claims: #{inspect(claims)}")
```
Comment (jonas-martinez) [inline lib/guardian/app_guardian.ex]:
```suggestion
        Logger.error("#{__MODULE__} Unknown token type for claims: #{inspect(claims)}")
```

File: lib/schema/json_shemata.ex [hunk 19]
```diff
@@ -17,10 +19,13 @@ defmodule ApplicationRunner.JsonSchemata do
   end
 
   def start_link(_) do
+    Logger.debug("#{__MODULE__} Start")
     GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
   end
 
-  # Server (callbacks)
+  @doc """
+    During init we load root component from component_api (root defined in @component_api_directory).
```
Comment (jonas-martinez) [inline lib/schema/json_shemata.ex]:
```suggestion
    During init we load the root component from component_api (root defined in @component_api_directory).
```

File: lib/session/dynamic_supervisor.ex [hunk 20]
```diff
@@ -49,10 +57,23 @@ defmodule ApplicationRunner.Session.DynamicSupervisor do
...
 
       pid ->
         DynamicSupervisor.terminate_child(get_full_name(env_id), pid)
+        |> case do
+          :ok ->
+            Logger.debug("Session not found for env_id: #{env_id}, session_id: #{session_id}")
+
+          {:error, :not_found} ->
+            Logger.warning(
+              "#{__MODULE__} DynamicSupervisor cannot found Session for env_id: #{env_id}, session_id: #{session_id}"
```
Comment (jonas-martinez) [inline lib/session/dynamic_supervisor.ex]:
```suggestion
              "#{__MODULE__} DynamicSupervisor could not find Session for env_id: #{env_id}, session_id: #{session_id}"
```

File: lib/session/events/on_session_start.ex [hunk 21]
```diff
@@ -14,20 +16,25 @@ defmodule ApplicationRunner.Session.Events.OnSessionStart do
...
+    Logger.debug("#{__MODULE__} run event for session_id: #{session_id}")
+
     case ApplicationRunner.EventHandler.send_session_event(
            session_id,
            @on_session_start_action,
            %{},
            %{}
          ) do
       :ok ->
+        Logger.debug("#{__MODULE__} succesfully run event for session_id: #{session_id}")
```
Comment (jonas-martinez) [inline lib/session/events/on_session_start.ex]:
```suggestion
        Logger.debug("#{__MODULE__} succesfully ran event for session_id: #{session_id}")
```

File: lib/session/events/on_user_first_join.ex [hunk 22]
```diff
@@ -28,12 +32,21 @@ defmodule ApplicationRunner.Session.Events.OnUserFirstJoin do
              %{},
              %{}
            ) do
+      Logger.debug(
+        "#{__MODULE__} succesfully run event for session_id: #{session_id} and user: #{user_id}"
```
Comment (jonas-martinez) [inline lib/session/events/on_user_first_join.ex]:
```suggestion
        "#{__MODULE__} succesfully ran event for session_id: #{session_id} and user: #{user_id}"
```

File: lib/session/events/on_user_first_join.ex [hunk 23]
```diff
@@ -28,12 +32,21 @@ defmodule ApplicationRunner.Session.Events.OnUserFirstJoin do
...
+      )
+
       {:ok, :ok, {:continue, :stop_me}}
     else
       true ->
+        Logger.debug("#{__MODULE__} user already run onUserFirstJoin for id: #{user_id}")
```
Comment (jonas-martinez) [inline lib/session/events/on_user_first_join.ex]:
```suggestion
        Logger.debug("#{__MODULE__} user already ran onUserFirstJoin for id: #{user_id}")
```

File: lib/session/ui_builders/lenra_builder.ex [hunk 24]
```diff
@@ -357,12 +367,23 @@ defmodule ApplicationRunner.Session.UiBuilders.LenraBuilder do
...
         {:ok, [], ui_context}
 
       children ->
-        build_children_map(session_metadata, children, ui_context, view_uid)
+        try do
+          build_children_map(session_metadata, children, ui_context, view_uid)
+        rescue
+          e ->
+            Logger.error(
+              "#{__MODULE__} fail to build children map for session_metadata: #{inspect(session_metadata)} with children: #{inspect(children)} and error: #{inspect(e)}"
```
Comment (jonas-martinez) [inline lib/session/ui_builders/lenra_builder.ex]:
```suggestion
              "#{__MODULE__} failed to build children map for session_metadata: #{inspect(session_metadata)} with children: #{inspect(children)} and error: #{inspect(e)}"
```

**Review by jonas-martinez — CHANGES_REQUESTED** [review]

**Review by jonas-martinez — CHANGES_REQUESTED** [review]
Can you fix the CI ?

**Review by jonas-martinez — APPROVED** [review]

**Review by taorepoara — APPROVED** [review]

**Pull request merged**